apollo-federation/src/operation/mod.rs

```diff
@@ -2137,7 +2138,23 @@ impl SelectionSet {
             type_position: self.type_position.clone(),
             selections: Arc::new(SelectionMap::new()),
         };
-        expanded.merge_selections_into(expanded_selections.iter())?;
+        for selection in expanded_selections {
```

Comment:
couldn't we just `merge_selections_into` and then normalize whole `expanded` set instead of normalizing and merging individual selections?

Reply:
Good point. Actually, normalization here is not necessary after all. I'll remove unnecessary changes.